libs/components/modals/src/lib/modules/modal/modal.component.ts

```diff
@@ -184,56 +206,56 @@ export class SkyModalComponent implements AfterViewInit, OnDestroy {
 
   public ngAfterViewInit() {
     skyModalUniqueIdentifier++;
-    this.componentAdapter.handleWindowChange(this.elRef);
+    this.#componentAdapter.handleWindowChange(this.#elRef);
 
     // Adding a timeout to avoid ExpressionChangedAfterItHasBeenCheckedError.
     // https://stackoverflow.com/questions/40562845
-    this.windowRef.nativeWindow.setTimeout(() => {
-      this.componentAdapter.modalOpened(this.elRef);
+    this.#windowRef.nativeWindow.setTimeout(() => {
+      this.#componentAdapter.modalOpened(this.#elRef);
     });
 
-    this.dockService.setDockOptions({
+    this.#dockService.setDockOptions({
       location: SkyDockLocation.ElementBottom,
-      referenceEl: this.modalContentWrapperElement.nativeElement,
+      referenceEl: this.modalContentWrapperElement!.nativeElement,
       zIndex: 5,
     });
 
     /* istanbul ignore next */
-    if (this.mediaQueryService) {
-      this.mediaQueryService.observe(this.modalContentWrapperElement);
+    if (this.#mediaQueryService) {
+      this.#mediaQueryService.observe(this.modalContentWrapperElement!);
     }
   }
 
   public ngOnDestroy(): void {
     /* istanbul ignore next */
-    if (this.mediaQueryService) {
-      this.mediaQueryService.unobserve();
+    if (this.#mediaQueryService) {
+      this.#mediaQueryService.unobserve();
     }
   }
 
   public helpButtonClick() {
-    this.hostService.onOpenHelp(this.helpKey);
+    this.#hostService.onOpenHelp(this.helpKey);
   }
 
   public closeButtonClick() {
-    this.hostService.onClose();
+    this.#hostService.onClose();
   }
 
   public windowResize() {
-    this.componentAdapter.handleWindowChange(this.elRef);
+    this.#componentAdapter.handleWindowChange(this.#elRef);
   }
 
   public scrollShadowChange(args: SkyModalScrollShadowEventArgs): void {
     this.scrollShadow = args;
   }
 
   public viewkeeperEnabled(): boolean {
```

Comment:
This function is called from the template. Does this value change during the lifecycle of the component, and if so, should we be relying purely on external change detection triggers to notify the template? Again, this should be an Observable so the template can be notified when this changes if it's outside a change detection cycle.

Reply:
I'm not sure I follow. This is a case where we are listening for an event and calling a function based on that. Not where we are binding to a value and calling a function. I'm not sure that I understand your suggestion on this one.

Reply:
I was referring to `viewkeeperEnabled()`. This line is in `modal.component.html`:

```
'sky-modal-viewkeeper': viewkeeperEnabled()
```

Reply:
We'll address in a future change.